tools/mavfft_pid.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+
+'''
+fit estimate of PID oscillations
+'''
+from __future__ import print_function
+
+import numpy
+import pylab
+
+from argparse import ArgumentParser
+parser = ArgumentParser(description=__doc__)
+parser.add_argument("--condition", default=None, help="select packets by condition")
+parser.add_argument("--sample-rate", dest='sample_rate', type=int, default=400, help="sample rate of PID values")
+parser.add_argument("logs", metavar="LOG", nargs="+")
+
+args = parser.parse_args()
+
+from pymavlink import mavutil
+
+def fft(logfile):
+    '''display fft for PID data in logfile'''
+
+    print("Processing log %s" % filename)
+    mlog = mavutil.mavlink_connection(filename)
+    sample_rate = args.sample_rate
+
+    data = {'PIDR.rate' : 400,
+            'PIDP.rate' : 400,
+            'PIDY.rate' : 400, }
+
+    for gyr in ['PIDR','PIDP', 'PIDY']:
+        for ax in ['P', 'I', 'D']:
+            data[gyr+'.'+ax] = []
+
+    # now gather all the data
+    while True:
+        m = mlog.recv_match(condition=args.condition)
+        if m is None:
+            break
+        type = m.get_type()
+        if type.startswith("PIDR"):
```

Comment:
Why not direct equivalence?  Could be rather faster if there's a checksum associated with the strings, for example.

Reply:
My python is like my french - limited and broken. I tend to stick with things I've copied and pasted. Anyway fixed now.